packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -183,7 +195,32 @@ class ${class_name}:
             Stream of measurement output.
         """
         parameter_values = [${measure_api_parameters}]
-        request = self._create_measure_request(parameter_values)
-
-        for response in self._get_stub().Measure(request):
-            yield self._deserialize_response(response)
+        with self._initialization_lock:
+            if self._measure_response is not None:
+                raise RuntimeError("A measure call is already in progress.")
```

Comment:
Elaborating on the error message could help users understand how to avoid this error.

```suggestion
                raise RuntimeError("A measurement is currently in progress. Concurrent measurement requests are not allowed.")
```
or

```suggestion
                raise RuntimeError("A measurement is currently in progress. To make concurrent measurement requests, please create a new client instance.")
```

Reply:
updated to `raise RuntimeError("A measurement is currently in progress. To make concurrent measurement requests, please create a new client instance.")`